app/logic/Mail/EWS/EWSAccount.ts

```diff
@@ -526,6 +527,12 @@ export class EWSAccount extends MailAccount {
         this.folderMap.set(folder.FolderId.Id, ewsFolder);
       }
     }
+    // Iterate from deepest to shallowest
+    for (let folder of this.getAllFolders().reverse()) {
+      if (!this.folderMap.has(folder.id)) {
```

Comment:
Don't delete, if we accidentally have 2 folder objects for the same folder id.
**Should** not happen, but if it does, we shouldn't destroy the folder. Always opt for safe.
I already commited the change.

Reply:
Should not happen? Actually it always happens when you move a folder... at least the folder still "works" in its previous location I guess... I'll see whether I can detect this case and move the folder.